### PR TITLE
Expose `ClassDB::is_enum_bitfield` as `ClassDB::is_class_enum_bitfield`

### PR DESCRIPTION
**Pull request by Naros**
Currently, there isn't a way for plug-in authors to resolve whether a given class-defined enum is a bitfield.  Having this knowledge exposed allows plug-in authors to render UIs and handle this specific case differently from normal enum types.

### CODE REVIEW

**Review by fire — APPROVED** [review]
Did not review the code, but enhancements helping gdextension authors have a better quality of life is a worthy goal.

File: doc/classes/ClassDB.xml [hunk 1]
```diff
@@ -192,6 +192,15 @@
 				Returns whether this [param class] is enabled or not.
 			</description>
 		</method>
+		<method name="is_class_enum_bitfield" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="class" type="StringName" />
+			<param index="1" name="enum" type="StringName" />
+			<param index="2" name="no_inheritance" type="bool" default="false" />
+			<description>
+				Returns whether the [param enum] is defined as a bitfield in [param class] or its ancestry classes.
```
Comment (dsnopek) [inline doc/classes/ClassDB.xml:201]: "ancestry" sounds weird to me here. Maybe:
```suggestion
				Returns whether the [param enum] is defined as a bitfield in [param class] or its ancestor classes.
```

**Review by dsnopek — APPROVED** [review]
This makes sense to me! The code looks good. Thanks!

File: doc/classes/ClassDB.xml (hunk 1 above)
Comment (AThousandShips) [reply doc/classes/ClassDB.xml:201]:
```suggestion
				Returns whether the [param enum] is defined as a bitfield in [param class] (or its ancestry if [param no_inheritance] is [code]false[/code]).
```

Comment (Naros) [reply doc/classes/ClassDB.xml:201]: I believe I took this from the other methods in ClassDB, should we adjust the other descriptions here or should be do that in a separate PR entirely?

Comment (dsnopek) [reply doc/classes/ClassDB.xml:201]: Ah, if it's in other places, then a separate PR probably makes sense

Comment (Naros) [reply doc/classes/ClassDB.xml]: Perfect, updated the newly added method only, thanks to you both 👍

File: doc/classes/ClassDB.xml [hunk 2]
```diff
@@ -192,6 +192,15 @@
 				Returns whether this [param class] is enabled or not.
 			</description>
 		</method>
+		<method name="is_class_enum_bitfield" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="class" type="StringName" />
+			<param index="1" name="enum" type="StringName" />
+			<param index="2" name="no_inheritance" type="bool" default="false" />
+			<description>
+				Returns whether the [param enum] is defined as a bitfield in [param class] or its ancestor classes.
```
Comment (AThousandShips) [inline doc/classes/ClassDB.xml:201]: I'd say it should still follow that style suggested as this doesn't clarify the option, meant to fix the other cases later, this should be made clear from the start

Comment (Naros) [reply doc/classes/ClassDB.xml:201]: So would you agree on this description @AThousandShips 
```
Returns whether [param class] or its ancestor classes has an enum called [param enum] that is a bitfield.
```

Comment (AThousandShips) [reply doc/classes/ClassDB.xml:201]: It should mention that `no_inheritance` controls things, like my suggestion following the general form of those cases

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**